packages/office-ui-fabric-react/src/components/Modal/Modal.base.tsx

```diff
@@ -221,7 +221,6 @@ export class ModalBase extends BaseComponent<IModalProps, IDialogState> implemen
         <Layer {...mergedLayerProps}>
           <Popup
             role={isModeless || !isBlocking ? 'dialog' : 'alertdialog'}
```

Comment:
>role={isModeless || !isBlocking ? 'dialog' : 'alertdialog'} [](start = 12, length = 59)

I'm glad we're removing aria-modal, but we should also remove the role.... In reality the Modal should be wrapper that hosts things and isn't anything (like) by itself... it's a vehicle for hosting a dialog, not the dialog itself

Reply:
Makes sense, but I'd rather limit this PR to dealing with `aria-modal` as I believe it's separate from the application of role here, and would have to consider all components using and used by Modal. Do you know of any issues that capture this? We could make one, if not.

Reply:
dialog still carries aria-modal inherently (with the value false) so this change is just potentially making the markup of the role worse

Reply:
Popup contains a FocusTrapZone, which would have the `aria-modal` property. I'm also concerned with having `aria-modal` on multiple layers unnecessarily, and `FocusTrapZone` can't know it's being used as a dialog.

What if I just close this PR and resolve the issue by making it the responsibility of users to pass `aria-modal` to FocusTrapZone?

Reply:
I think it should be the responsibility of the user to add aria-modal, but case like Dialog and Panel feel like they should have it by default

...and until modal stops having the ability to be a leaf component I guess it should have it it as well... but we're back to square one... If the modal couldn't be a leaf component I'm pretty sure it would simplify things a lot. Alternatively, if we could detect that the modal is being used as a leaf component they we could make this work

We need to make we aren't introducing accessibility bugs for users that are relying on the components to be marked up already (as a side effect of us trying to clean things up for accessibility... This might all have to be done together so no regressions are introduced)

Reply:
Yeah, my intention was that they would still have it, since these components are using `FocusTrapZone`, and thus still would have `aria-modal`. In any case, there is no effect on navigation currently, but I do not want to misapply it and have closed this PR for now.